File: WebContent/Components/Hospital.js
$(document).ready(function() {

    $("#alertSuccess").hide();
    $("#alertError").hide();

});

// Save
$(document).on("click", "#btnSave", function(event) {

    // Clear alerts
    $("#alertSuccess").text("");
    $("#alertSuccess").hide();
    $("#alertError").text("");
    $("#alertError").hide();

    // Form validation
    var status = validateHospitalForm();
    if (status != true) {
        $("#alertError").text(status);
        $("#alertError").show();
        return;
    }

    // If valid
    var type = ($("#hidHospitalIDSave").val() == "") ? "POST" : "PUT";

    $.ajax(
        {
            url : "hospitalAPI",
            type : type,
            data : $("#formHospital").serialize(),
            dataType : "text",
            complete : function(response, status)
            {
                onUserSaveComplete(response.responseText, status);
            }
        });

});

function onUserSaveComplete(response, status) {

    if (status == "success") {

        var resultSet = JSON.parse(response);

        if (resultSet.status.trim() == "success") {

            $("#alertSuccess").text("Successfully saved.");
            $("#alertSuccess").show();
            $("#divUsersGrid").html(resultSet.data);

        } else if (resultSet.status.trim() == "error") {

            $("#alertError").text(resultSet.data);
            $("#alertError").show();

        }
    } else if (status == "error") {

        $("#alertError").text("Error while saving.");
        $("#alertError").show();

    } else {

        $("#alertError").text("Unknown error while saving..");
        $("#alertError").show();

    }

    $("#hidHospitalIDSave").val("");
    $("#formUser")[0].reset();

}
// UPDATE==========================================
$(document).on("click", ".btnUpdate", function(event) {
    $("#hidHospitalIDSave").val($(this).closest("tr").find('#hidHospitalIDUpdate').val());
    $("#Hospital_Name").val($(this).closest("tr").find('td:eq(0)').text());
    $("#Hospital_Address").val($(this).closest("tr").find('td:eq(1)').text());
    $("#Hospital_City").val($(this).closest("tr").find('td:eq(2)').text());
    $("#Hospital_Phone").val($(this).closest("tr").find('td:eq(3)').text());
    $("#Hospital_Email").val($(this).closest("tr").find('td:eq(4)').text());
    $("#Hospital_Description").val($(this).closest("tr").find('td:eq(5)').text());
    $("#Open_Hours").val($(this).closest("tr").find('td:eq(6)').text());
});

//Remove
$(document).on("click", ".btnRemove", function(event)
{
    $.ajax(
        {
            url : "hospitalAPI",
            type : "DELETE",
            data : "DoctorID=" + $(this).data("DoctorID"),
            dataType : "text",
            complete : function(response, status)
            {
                onUserDeleteComplete(response.responseText, status);
            }
        });
});

function onHospitalDeleteComplete(response, status) {

    if (status == "success") {

        var resultSet = JSON.parse(response);

        if (resultSet.status.trim() == "success") {

            $("#alertSuccess").text("Successfully deleted.");
            $("#alertSuccess").show();
            $("#divHospitalGrid").html(resultSet.data);

        } else if (resultSet.status.trim() == "error") {

            $("#alertError").text(resultSet.data);
            $("#alertError").show();

        }

    } else if (status == "error") {

        $("#alertError").text("Error while deleting.");
        $("#alertError").show();

    } else {

        $("#alertError").text("Unknown error while deleting..");
        $("#alertError").show();

    }

}



// CLIENTMODEL=========================================================================
function validateHospitalForm() {
    
    if ($("#Hospital_Name").val().trim() == "") {
        return "Insert Hospital name.";
    }
    
    if ($("#Hospital_Address").val().trim() == "") {
        return "Insert Hospital Address.";
    }
    

    if ($("#Hospital_City").val().trim() == "") {
        return "Insert Hospital City.";
    }

    if ($("#Hospital_Phone").val().trim() == "") {
        return "Insert Hospital Phone.";
    }
    
    if ($("#Hospital_Email").val().trim() == "") {
        return "Insert Hospital email.";
    }
    
    if ($("#Hospital_Description").val().trim() == "") {
        return "Insert Hospital Description.";
    }
    
    if ($("#Open_Hours").val().trim() == "") {
        return "Insert Hospital Open_Hours.";
    }
    return true;
}

File: src/com/java/hospital.java
package com.java;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.Statement;



import com.resources.connect;

public class hospital {
	
		connect DbObj = new connect();
		
		//Read Hospitals Details
		public String readHospitals() {  
			String output = "";  


			try {  
				Connection con = DbObj.connectMethod();
				if (con == null)  {   
					return "Error while connecting to the database for reading.";  
				} 

			// Prepare the html table to be displayed   
				output = "<table border=\"1\"><tr>"
						+ "<th>Hospital_Name</th>"
						+ "<th>Hospital_Address</th>"
						+ "<th>Hospital_City</th>"
						+ "<th>Hospital_Phone</th>"
						+ "<th>Hospital_Email</th>"
						+ "<th>Hospital_Description</th>"
						+ "<th>Open_Hours</th>"
						+ "<th>UPDATE</th>"
						+ "<th>REMOVE</th>"	
						+ "</tr>";
			  String query = "select * from hospitals";   
			  Statement stmt = con.createStatement();   
			  ResultSet rs = stmt.executeQuery(query); 

			  // iterate through the rows in the result set   
			  while (rs.next())   {    
				  String Hospital_ID = Integer.toString(rs.getInt("Hospital_ID"));    
				  String Hospital_Name = rs.getString("Hospital_Name");    
				  String Hospital_Address = rs.getString("Hospital_Address");    
				  String Hospital_City = rs.getString("Hospital_City");    
				  String Hospital_Phone = rs.getString("Hospital_Phone"); 
				  String Hospital_Email = rs.getString("Hospital_Email");
				  String Hospital_Description = rs.getString("Hospital_Description");
				  String Open_Hours = Integer.toString(rs.getInt("Open_Hours"));  

			   // Add into the html table    
				
				  
			  output += "<tr><td><input id='hidHospitalIDUpdate' name='hidHospitalIDUpdate' type='hidden' value='" + Hospital_ID + "'></td>";
			  output += "<tr><td>" + Hospital_Name + "</td>";       
			  output += "<td>" + Hospital_Address + "</td>";
			  output += "<td>" + Hospital_City + "</td>";    
			  output += "<td>" + Hospital_Phone + "</td>"; 
			  output += "<td>" + Hospital_Email + "</td>";    
			  output += "<td>" + Hospital_Description + "</td>";
			  output += "<td>" + Open_Hours + "</td>"; 

//			   // buttons  
			  
			  output  += "<td><input name='btnUpdate' type='button' value='Update' "
						+ "class='btnUpdate btn btn-secondary'></td>" + "<td><input name='btnRemove' type='button' value='Remove'"
								+ " class='btnRemove btn btn-danger' data-hospitalid= '" + Hospital_ID + "'>" + "</td></tr>";

			 
			/*  output += "<td><input name=\"btnUpdate\" type=\"button\"value=\"Update\" class=\"btn btn-warning btnUpdate\"></td>"
						+ "<td><form method=\"post\" action=\"hospital.jsp\">"
						+ "<input name=\"btnRemove\" type=\"submit\" value=\"Remove\"class=\"btn btn-danger\">"
						+ "<input name=\"hidHospitalIDDelete\" type=\"hidden\" value=\"" + Hospital_ID + "\">"
						+ "</form></td></tr>"; */

     		
			  } 

			  con.close(); 

			  // Complete the html table   
			  output += "</table>"; 
			}
			catch (Exception e) {  
				output = "Error while reading the Hospital data.";  
				System.err.println(e.getMessage()); 
			}

			return output;
		}	
		public String insertHospital(String hos_id,String name, String address, String city, String phone, String email, String desc, String hours) {
			String output = "";

			try {
				Connection con = DbObj.connectMethod();

				if (con == null) {
					return "Error while connecting to the database";
				}

				// create a prepared statement   
				String query = " insert into hospitals ('Hospital_ID',`Hospital_Name`,`Hospital_Address`,`Hospital_City`,`Hospital_Phone`,`Hospital_Email`,`Hospital_Description`,`Open_Hours`)"+" values (?, ?, ?, ?, ?, ?, ?, ?)";

				PreparedStatement preparedStmt = con.prepareStatement(query);

				// binding values 
				preparedStmt.setInt(1, 0);   
				preparedStmt.setString(2, name);   
				preparedStmt.setString(3, address);    
				preparedStmt.setString(4, city);
				preparedStmt.setString(5, phone);
				preparedStmt.setString(6, email);
				preparedStmt.setString(7, desc);
				preparedStmt.setInt(8, Integer.parseInt(hours));  

				//execute the statement   
				preparedStmt.execute();   
				con.close(); 

				String newHospital = readHospitals(); output = "{\"status\":\"success\", \"data\": \"" +    newHospital + "\"}"; 

			}
			catch (Exception e) {   
				output = "{\"status\":\"error\", \"data\":     \"Error while inserting the Hospitals.\"}";
		  
				System.err.println(e.getMessage());  
			} 

			 return output; 
		}
		
		public String updateHospitals(String hos_id, String name, String address, String city, String phone, String email, String desc, String hours)  {   
			String output = ""; 
		 
		  try   {   
			  Connection con = DbObj.connectMethod();
		 
			  if (con == null)    {
				  return "Error while connecting to the database for updating."; 
			  } 
		 
		   // create a prepared statement    
		   String query = "UPDATE hospitals SET Hospital_Name=?,Hospital_Address=?,Hospital_City=?,Hospital_Phone=?,Hospital_Email=?,Hospital_Description=?,Open_Hours=?      "
		   		+ "			WHERE Hospital_ID=?"; 
		 
		   PreparedStatement preparedStmt = con.prepareStatement(query); 
		 
		   // binding values    
		   preparedStmt.setInt(1, 0); 
		   preparedStmt.setString(1, name);    
		   preparedStmt.setString(2, address);    
		   preparedStmt.setString(3, city);
		   preparedStmt.setString(4, phone);
		   preparedStmt.setString(5, email);
		   preparedStmt.setString(6, desc);
		   preparedStmt.setInt(7, Integer.parseInt(hours));
		   preparedStmt.setInt(8,Integer.parseInt(hos_id));
		 
		   // execute the statement    
		   preparedStmt.execute();    
		   con.close(); 
		 
		   String newHospital = readHospitals(); output = "{\"status\":\"success\", \"data\": \"" +    newHospital + "\"}"; 

		   output = "Updated successfully";   
		   }   catch (Exception e)   {  
			   
		    	output = "{\"status\":\"error\", \"data\":     \"Error while updating the Hospitals.\"}";

			    
			   System.err.println(e.getMessage());   
		   } 
		 
		  return output;  
		  }
		
		public String deleteHospitals(String hos_id) {  
			String output = ""; 
		 
		 try  {   
			 Connection con = DbObj.connectMethod();
		 
		  if (con == null)   {    
			  return "Error while connecting to the database for deleting.";   
		  } 
		 
		  // create a prepared statement   
		  String query = "delete from hospitals where Hospital_ID=?"; 
		 
		  PreparedStatement preparedStmt = con.prepareStatement(query); 
		 
		  
		  preparedStmt.setInt(1, Integer.parseInt(hos_id));       
		     
		  preparedStmt.execute();   
		  con.close(); 
		 
		  String newHospital = readHospitals(); output = "{\"status\":\"success\", \"data\": \"" +    newHospital + "\"}"; 

		  output = "Deleted successfully";  
		  }  catch (Exception e)  {  
			  
			  output = "{\"status\":\"error\", \"data\":     \"Sorry , Error while deleting the hospital.\"}";

			  
			  System.err.println(e.getMessage());  
			  
		 } 
		 
		 return output; 
		 }
	}


